docs/docs/Concepts/concepts-components.md

```diff
@@ -0,0 +1,209 @@
+
+**Input Definition:** Each input (like `IntInput` or `DataInput`) specifies an input's type, name, and display properties, which appear as configurable fields in the component's UI panel.
+
+**Methods:** Components have methods or functions that handle their functionality. This component has two methods.
+`get_data_input` retrieves the text data to be split from the component's input. This makes the data available to the class.
+`build_text_splitter` creates a `RecursiveCharacterTextSplitter` object by calling its parent class's `build` method. The text is split with the created splitter and passed to the next component.
+When used in a flow, this component:
+
+1. Displays its configuration options in the UI.
+2. Validates user inputs based on the input types.
+3. Processes data using the configured parameters.
+4. Passes results to the next component.
+
+## Freeze Path
+
+After a component runs, **Freeze Path** locks the component's previous output state to prevent it from re-running.
+
+If you’re expecting consistent output from a component and don’t need to re-run it, click **Freeze Path**. 
+
+Enabling **Freeze Path** freezes all components upstream of the selected component.
+
+If you only want to freeze a single component, select **Freeze** instead.
+
+A <Icon name="Snowflake" aria-label="Snowflake"/> icon appears on all frozen components.
+
+## Additional component options
+
+Click <Icon name="Ellipsis" aria-label="Horizontal ellipsis" /> **All** to see additional options for a component.
+
+To modify a component's name or description, double-click in the **Name** or **Description** fields. Component descriptions accept markdown syntax.
+
+### Component shortcuts
+
+The following keyboard shortcuts are available when a component is selected.
+
+| Menu Item | Mac Shortcut | Description |
+|-----------|----------|-------------|
+| Code | ⌘ + C | Opens the code editor for the component. |
+| Advanced | ⌘ + A | Opens advanced settings for the component. |
+| Save | ⌘ + S | Saves the current state of the component to Saved components in the sidebar. |
+| Duplicate | ⌘ + D | Creates a duplicate of the component. |
+| Copy | ⌘ + C | Copies the selected component. Paste it in the workspace with ⌘ + V. |
+| Docs | ⌘ + D | Opens related documentation. |
+| Minimize | ⌘ + Q | Minimizes the current component. |
+| Freeze | ⌘ + F | Freezes the current component state. |
+| Freeze Path | ⌘ + Shift + F | Freezes the current component state and all upstream components. |
+| Download | ⌘ + D | Downloads the current component as a JSON file. |
+| Delete | ⌘ + ⌫ | Deletes the component. |
+
+## Group components in the workspace
+
+Multiple components can be grouped into a single component for reuse. This is useful when combining large flows into single components (like RAG with a vector database, for example) and saving space.
+
+1. Hold **Shift** and drag to select components.
+2. Select **Group**.
+3. The components merge into a single component.
```

Comment:
The user does not perform this step so it should not be a numbered step. Make this a sentence below step 2.